File: Top_Interview_150/Hashmap/python/383.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:

        ransomNote_case = {c: ransomNote.count(c) for c in ransomNote}
        for c in magazine:
            if c in ransomNote_case:
                ransomNote_case[c] -= 1

        print(ransomNote_case)

        return all(c <= 0 for c in ransomNote_case.values())


solution = Solution()
print(solution.canConstruct(ransomNote="aa", magazine="ab"))

File: Top_Interview_150/Hashmap/cpp/383.cpp
#include <string>
#include <unordered_map>

using namespace std;

class Solution
{
public:
    bool canConstruct(string ransomNote, string magazine)
    {
        unordered_map<char, int> ransomNote_case;
        for (char c : ransomNote)
            ransomNote_case[c] += 1;

        for (char c : magazine)
            if (ransomNote_case.find(c) != ransomNote_case.end())
                ransomNote_case[c]--;

        for (const auto &[key, value] : ransomNote_case)
            if (value > 0)
                return false;

        return true;
    }
};